### PR TITLE
validation: port some of the functionality to reana-commons

### PR DESCRIPTION
**Pull request by audrium**
closes https://github.com/reanahub/reana-server/issues/442

TODO:
- [ ] Double test all the edge cases 
- [ ] Add changelog message
- [ ] Port and fix tests

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,9 +1,10 @@
 Changes
 =======
 
-Version 0.8.2 (UNRELEASED)
+Version 0.9.0 (UNRELEASED)
```
Comment (tiborsimko) [inline CHANGES.rst:4]: You can keep the 0.8.2 line that we might release for 0.8 bug fixes.

So just add a new 0.9.0 section for the current change.

**Review by tiborsimko — APPROVED** [review]
